### PR TITLE
feat(alerts): Refactor Alert Settings to FC

### PR DESCRIPTION
**Pull request by scttcper**
Refactor Alert Settings to be a functional component. Adds a test for enabling an integration from this page, which is kinda a weird piece of functionality.

### CODE REVIEW

File: static/app/types/integrations.tsx [hunk 1]
```diff
@@ -467,11 +467,11 @@ export type PluginNoProject = {
   // TODO(ts)
   status: string;
   type: string;
-  altIsSentryApp?: boolean;
+  altIsSentryApp?: boolean | null;
```
Comment (scttcper) [inline static/app/types/integrations.tsx:470]: just from copy pasting the webhook config from the api, these types were wrong

File: static/app/views/settings/projectAlerts/settings.tsx [hunk 2]
```diff
@@ -44,17 +44,30 @@ function ProjectAlertSettings({canEditRule, params}: ProjectAlertSettingsProps)
...
   );
 
-  if (!isProjectLoading && !project) {
-    return <LoadingError onRetry={refetchProject} />;
+  if ((!isProjectLoading && !project) || isPluginListError || isProjectError) {
+    return (
+      <LoadingError
+        onRetry={() => {
+          refetchProject();
+          refetchPluginList();
```
Comment (JonasBa) [inline static/app/views/settings/projectAlerts/settings.tsx:67]:
```suggestion
          isProjectError && refetchProject();
          isPluginError && refetchPluginList();
```

File: static/app/views/settings/projectAlerts/settings.tsx [hunk 3]
```diff
@@ -1,168 +1,176 @@
...
+
+      {isProjectLoading || isPluginListLoading ? (
+        <LoadingIndicator />
+      ) : (
+        <Fragment>
+          <Form
+            saveOnBlur
+            allowUndo
+            initialData={{
+              subjectTemplate: project.subjectTemplate,
```
Comment (malwilley) [inline static/app/views/settings/projectAlerts/settings.tsx:132]: Does the project cache get updated after these changes are made? I'm wondering what happens if you modify a setting here, go to another page, then come back. The form might see the old state and not update when the project refetch occurs.

Comment (scttcper) [reply static/app/views/settings/projectAlerts/settings.tsx:133]: set cacheTime to 0

**Review by malwilley — APPROVED** [review]

**Pull request merged**